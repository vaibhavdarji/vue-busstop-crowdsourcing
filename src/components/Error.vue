<template>
  <v-layout>
    <v-flex xs8>
      <v-list-tile-title :class="Constants.THEME.ERROR_TEXT_CLASS">
        {{msg}}
        </v-list-tile-title>
    </v-flex>
    <v-flex xs4>
      <custom-button @onClick="onRefresh()">
          Refresh
      </custom-button>
    </v-flex>
  </v-layout>
</template>

<script>
import Constants from '@/constants';
import CustomButton from '@/components/Button.vue';
export default {
  name: 'CustomError',
  props: {
    msg: {
      type: String,
      default: 'Not able to perform the operation at moment! Please try again later.'
    }
  },

  components: {
    CustomButton
  },

  computed: {
    Constants () {
      return Constants;
    }
  },

  methods: {
    onRefresh () {
      this.$emit('onErrorBtnClick');
    }
  }
}
</script>

