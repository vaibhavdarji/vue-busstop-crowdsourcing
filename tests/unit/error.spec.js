import { shallowMount, mount } from "@vue/test-utils";
import Error from "@/components/Error.vue";
import Vue from "vue";
import Vuetify from "vuetify";

Vue.use(Vuetify);

describe("Error.vue", () => {
  it("should match snapshot", () => {
    const wrapper = shallowMount(Error, {
      Vue
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should set msg prop and validate in dom content", () => {
    
    const wrapper = mount(Error, {
      Vue
    });
    const msg = 'Unable to connect to database';

    wrapper.setProps({
      msg
    });
    expect(wrapper.find(`.error--text`).text()).toContain(msg);
  });
});