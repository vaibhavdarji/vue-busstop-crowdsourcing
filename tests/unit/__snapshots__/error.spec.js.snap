// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error.vue should match snapshot 1`] = `
<v-layout-stub
  tag="div"
>
  <v-flex-stub
    tag="div"
    xs8=""
  >
    <v-list-tile-title-stub
      class="error--text title font-weight-bold"
    >
      
      Not able to perform the operation at moment! Please try again later.
      
    </v-list-tile-title-stub>
  </v-flex-stub>
   
  <v-flex-stub
    tag="div"
    xs4=""
  >
    <custom-button-stub
      btntype="button"
    >
      
        Refresh
    
    </custom-button-stub>
  </v-flex-stub>
</v-layout-stub>
`;
